### ISSUE TITLE
How to enable SSL on Synology?

### ISSUE THREAD

**Issue by Raptor05** [jboxberger/synology-gitea-jboxberger#1]
Hey,
first of all: Thanks for the great package!

I have a question about setting up SSL. The readme explains that SSL can be enabled using
`sudo./var/packages/synology-gitea-jboxberger/scripts/gitea-ssl enable`. 
Unfortunately I can't execute **sudo** via SSH (I get a "command not found" despite the latest DSM version and root). What do I have to do to set up SSL and deposit a certificate?
Thank you in advance.

**Comment by jboxberger** [comment]
Then your user seems not to be in sudo group. You can execute it as root without sudo. if ./ does not work for you, you can try sh /var/... as root.

Hope this helps. 

PS: If you do not need SSL in your LAN then is is better to go without SSL. SSL is always an overhead and can bring many problems especially for beginners.

Kind Regards

**Comment by Raptor05** [comment]
Thanks for the fast response.
I've now tried it with the sh command and it works now.
The problem is, I can't reach the site using https... I've forwarded the port in the router, but I always get an "ERR_CONNECTION_REFUSED" in the browser. With HTTP port it works. Is it necessary to configure something in gitea? These are my configs:
![gitea_config](https://user-images.githubusercontent.com/4174662/41488385-6f891faa-70ec-11e8-8323-d293bbea24e0.png)

PS: I need the SSL because I want to reach the site also outside the local network..

Best regards.

**Comment by jboxberger** [comment]
Hi,

looks like a messed up port configuration.

You need to specify [incoming-docker-port] to [container-port] and this has to match wit yout gitea config. It also looks like you're trying to route HTTP over the Port 2222 which is the SSH Port. Thats what i meant with problems. Please check your ports are fine.

![grafik](https://user-images.githubusercontent.com/1545845/41489257-a7420364-70ef-11e8-9263-0b52d3a4daa8.png)

![grafik](https://user-images.githubusercontent.com/1545845/41489379-264f9dd8-70f0-11e8-9854-9663fd287a7e.png)

Kind Regards

**Comment by jboxberger** [comment]
Please note that you will get your encrypted connection but wihout the "green lock". This is because of the SELF-SIGNED-CERT, you can replace this wih a Letsencrypt one but i can not support you hereby. 

![grafik](https://user-images.githubusercontent.com/1545845/41489537-a187293a-70f0-11e8-8a7b-5e3c87e3185f.png)

Kind Reagrds

**Comment by Raptor05** [comment]
Unfortunately, nothing works anymore now. I've written the local gitea-docker port in the gitea config and then tried to call both the docker-container port and the local port, nothing works.
This is my docker port config:
![screen1_docker_config](https://user-images.githubusercontent.com/4174662/41490736-a73e0e70-70f5-11e8-8c79-cc9056b2c823.png)

And this is my gitea app.ini config:

> [server]
> KEY_FILE=/data/certs/key.pem
> CERT_FILE=/data/certs/cert.pem
> PROTOCOL=https
> APP_DATA_PATH    = /data/gitea
> SSH_DOMAIN       =%(DOMAIN)s
> HTTP_PORT        = 3001
> ROOT_URL         =%(PROTOCOL)s://%(DOMAIN)s:%(HTTP_PORT)s/
> DISABLE_SSH      = false
> SSH_PORT         = 3001
> DOMAIN           = hereismydomain.myds.me
> LFS_START_SERVER = true
> LFS_CONTENT_PATH = /data/gitea/lfs
> LFS_JWT_SECRET   = asd213
> OFFLINE_MODE     = false

PS: Why don't you use a top level domain in your config? How can this work?

**Comment by jboxberger** [comment]
This is easy, i do not use a TLD because i will never put my private repository to the Internet. Never ever ever. I use it for my local work at home and put public code on github. 

But why do you try to use the same port on SSH and HTTP?!!?? 
They have to be different. This can not WORK!

Your Configuration: 
HTTP_PORT = **_3001_**
SSH_PORT = 3001

My Configuration:
HTTP_PORT = **_3000_**
SSH_PORT = 3001

**Fix your port double use.**  

If you forward the Port 2221 (by the way your SSH Port) to your NAS and your Gitea is listenening to "HTTP_PORT = 3001", how this should work? You need to forward the same port in your router as the Gitea HTTP_PORT.

First of all get your instance running an then think about forwarding.


Kind Regards

**Comment by Raptor05** [comment]
I've tried it before this way, but it hasn't worked. So I changed the configuration, because I thought Listen-Port and Port are SSH Port and HTTP port. Why do you think my SSH port is 2221? I've now changed it back to http 3000 and ssh 3001 but it still doesn't work on SSH (only HTTP). Neither externally nor local.

**Comment by jboxberger** [comment]
I meant the  2222 PORT as your SSH PORT. 

Your SSH issue is a Gitea thing. I is not even a issue its just a common ssh based authentication same as in Gitlab or Github. You need to create ssh key, upload it to your Gitea and configure your system using the ssh key for your git server as auth... but i am not a Gitea maintainer and also not your LAN Routing expert. 

Hope you can understand that i can't an will not give support for ssh based authentication. The internet is full of examples. And there are even Help Links in Gitea itself.
![grafik](https://user-images.githubusercontent.com/1545845/41628483-09a980f8-7425-11e8-9d83-127896ebbb29.png)

Or Google:
https://discuss.gogs.io/t/how-to-config-ssh-settings/34

If you have any Problem with installing, runng or updating the docker container, i am your man. Anything inside the container is not my department. Sorry for sying that this way but i have to make this clear since i am doing this for free in my free time and can not give anyone support for anything.

Where is the Problem to use HTTPS also for the checkout the same way you use for external connections? Using both protocolls is technicallly duplication of the attaking vectors and is not neccesary.

Kidn Regards

**Comment by Raptor05** [comment]
I have now deactivated the SSH part again and deleted the port forwarding. I think I'll take your advice and use Gitea only locally. I have now also activated the local domain of the Synology device, which unfortunately only works sporadically (but that's another story). For the external access I will then switch to VPN (although this does not work in the UMTS network, it is a tragedy). 

I thank you anyway for your support and your detailed explanations in your spare time. I think this issue can now be marked as "done".

Best regards
